### PR TITLE
Empire readiness refactoring

### PR DESCRIPTION
**Pull request by o01eg**
Moves readiness status into empires itself.

### CODE REVIEW

File: client/ClientApp.cpp [hunk 1]
```diff
@@ -103,16 +103,12 @@ const std::map<int, PlayerInfo>& ClientApp::Players() const
...
-
-std::map<int, Message::PlayerStatus>& ClientApp::EmpireStatus()
-{ return m_empire_status; }
-
 void ClientApp::SetEmpireStatus(int empire_id, Message::PlayerStatus status) {
     if (empire_id == ALL_EMPIRES)
         return;
-    m_empire_status[empire_id] = status;
+    Empire *empire = m_empires.GetEmpire(empire_id);
+    if (empire != nullptr)
```
Comment (geoffthemedio) [inline client/ClientApp.cpp]:
````
if (auto* empire = m_empires.GetEmpire(empire_id))
    empire->SetReady(status == Message::WAITING);
````

File: client/ClientApp.cpp [hunk 2]
```diff
@@ -103,16 +103,12 @@ const std::map<int, PlayerInfo>& ClientApp::Players() const
...
 { return m_player_info; }
 
-const std::map<int, Message::PlayerStatus>& ClientApp::EmpireStatus() const
-{ return m_empire_status; }
-
-std::map<int, Message::PlayerStatus>& ClientApp::EmpireStatus()
-{ return m_empire_status; }
-
 void ClientApp::SetEmpireStatus(int empire_id, Message::PlayerStatus status) {
     if (empire_id == ALL_EMPIRES)
```
Comment (geoffthemedio) [inline client/ClientApp.cpp]: can also remove this check since `ALL_EMPIRES` should never be a valid empire id

File: server/ServerApp.cpp [hunk 3]
```diff
@@ -705,11 +705,13 @@ void ServerApp::NewGameInitConcurrentWithJoiners(
             m_player_empire_ids[player_id_and_setup.second.m_player_id] = empire_id;
 
         // add empires to turn processing
-        if (GetEmpire(empire_id))
+        Empire *empire = GetEmpire(empire_id);
```
Comment (geoffthemedio) [inline server/ServerApp.cpp]: same here

File: server/ServerApp.cpp [hunk 4]
```diff
@@ -2034,9 +2004,9 @@ void ServerApp::UpdatePartialOrders(int empire_id, const OrderSet& added, const
 }
 
 void ServerApp::RevokeEmpireTurnReadyness(int empire_id) {
-    const auto& psgd = m_turn_sequence[empire_id];
-    if (psgd)
-        psgd->m_ready = false;
+    Empire* empire = GetEmpire(empire_id);
```
Comment (geoffthemedio) [inline server/ServerApp.cpp]: ...

**Comment by o01eg** [comment]
Fixed and rebased.

**Pull request merged**